### PR TITLE
Added test for example at CBD definition. Fixes #1914.

### LINKED ISSUE #1914: Include Concise Bounded Description example in the tests for `Graph.cbd`
From a brief glance at `Graph.cbd` I'm unsure if it will be able to reproduce [the example from the Concise Bounded Description](https://www.w3.org/Submission/CBD/#example) definition, but regardless this example should be included in our tests so we can be sure it passes.

### PR DESCRIPTION
**Pull request by asejal**
Fixes #1914 

# Summary of changes

We have added a test for the example at the definition for Concise Bounded Definition [here](https://www.w3.org/Submission/CBD/#example). It has been converted to turtle format for uniformity. 

For the given RDF graph as input:
WhatsApp Image 2022-05-12 at 11 40 03 PM

Expected output for CBD of `http://example.com/aReallyGreatBook` corresponds to the following subgraph:
WhatsApp Image 2022-05-12 at 11 41 52 PM

The test passes. 
In collaboration with @ms1901.

### CODE REVIEW

File: test/test_graph/test_graph_cbd.py [hunk 1]
```diff
@@ -109,3 +109,71 @@ def testCbdReified(get_graph):
     )
 
     assert len(g.cbd(EX.R6)) == (3 + 5 + 5), "cbd() for R6 should return 12 triples"
+
+    # add example from Concise Bounded Description definition at https://www.w3.org/Submission/CBD/#example
+    g.parse(
```
Comment (aucampia) [inline test/test_graph/test_graph_cbd.py:114]: This should be in a separate test function.

File: test/test_graph/test_graph_cbd.py [hunk 2]
```diff
@@ -109,3 +109,71 @@ def testCbdReified(get_graph):
...
+                    dc:format     "application/pdf" ;
+                    dc:language   "en" ;
+                    dc:publisher  "Examples-R-Us" ;
+                    dc:rights     "Copyright (C) 2004 Examples-R-Us. All rights reserved." ;
+                    dc:title      "Another Great Book" ;
+                    dct:issued    "2004-05-03"^^xsd:date .
+
+            ex:john.jpg  a     foaf:Image ;
+                    dc:extent  "1234" ;
+                    dc:format  "image/jpeg" .
```
Comment (aucampia) [inline test/test_graph/test_graph_cbd.py:172]: I would prefer this rather go into a file in `test/data/spec/cbd/example_graph.rdf` as RDF XML (i.e. identical to what is [here](https://www.w3.org/Submission/CBD/#example)) and then place the `concise bounded description of http://example.com/aReallyGreatBook`  in `test/data/spec/cbd/example_graph_cbd.rdf`

File: test/test_graph/test_graph_cbd.py [hunk 3]
```diff
@@ -109,3 +109,71 @@ def testCbdReified(get_graph):
...
+        """,
+        format="turtle",
+    )
+
+    assert len(g.cbd(EX.aReallyGreatBook)) == (
+        21
+    ), "cbd() for aReallyGreatBook should return 21 triples"
```
Comment (aucampia) [inline test/test_graph/test_graph_cbd.py:179]: This is not sufficent, this should check that the resulting graph from `Graph.cbd` is isomorphic to and identical to (when discarding triples) the graph given in https://www.w3.org/Submission/CBD/#example

To do this you can use

[`GraphHelper.assert_isomorphic`](https://github.com/RDFLib/rdflib/blob/5e5dceaea453a5eb7fe2718f001c067e5be19c8e/test/utils/__init__.py#L221) and [`GraphHelper.assert_sets_equals(..., exclude_blanks=True)`](https://github.com/RDFLib/rdflib/blob/5e5dceaea453a5eb7fe2718f001c067e5be19c8e/test/utils/__init__.py#L191) for this.

**Review by aucampia — COMMENTED** [review]
Thank you very much for having a look at this @asejal - though please see the feedback.

File: test/test_graph/test_graph_cbd.py (hunk 3 above)
Comment (asejal) [reply test/test_graph/test_graph_cbd.py]: Thank you for the review.
We have completed the above requirements. We have created a separate function for testing the CBD example along with all the above mentioned asserts, and also added two separate data files.

**Comment by asejal** [comment]
pre-commit.ci autofix

File: test/test_graph/test_graph_cbd.py [hunk 4]
```diff
@@ -1,6 +1,13 @@
...
-
+import os
 from rdflib import Graph, Namespace
+from rdflib.term import URIRef
+from rdflib.compare import isomorphic
+from test.utils import GraphHelper
+from test.data import TEST_DATA_DIR
+
+EXAMPLE_GRAPH_FILE_PATH = os.path.join(*[TEST_DATA_DIR, "spec", "cbd", "example_graph.rdf"])
+EXAMPLE_GRAPH_CBD_FILE_PATH = os.path.join(*[TEST_DATA_DIR, "spec", "cbd", "example_graph_cbd.rdf"])
```
Comment (aucampia) [inline test/test_graph/test_graph_cbd.py]: This is causing some failure:

```console
  test/test_graph/test_graph_cbd.py:9: error: No overload variant of "join" matches argument type "List[object]"  [call-overload]
  test/test_graph/test_graph_cbd.py:9: note: Possible overload variants:
  test/test_graph/test_graph_cbd.py:9: note:     def join(__a, Union[str, PathLike[str]], *paths: Union[str, PathLike[str]]) -> str
  test/test_graph/test_graph_cbd.py:9: note:     def join(__a, Union[bytes, PathLike[bytes]], *paths: Union[bytes, PathLike[bytes]]) -> bytes
  test/test_graph/test_graph_cbd.py:10: error: No overload variant of "join" matches argument type "List[object]"  [call-overload]
  test/test_graph/test_graph_cbd.py:10: note: Possible overload variants:
  test/test_graph/test_graph_cbd.py:10: note:     def join(__a, Union[str, PathLike[str]], *paths: Union[str, PathLike[str]]) -> str
  test/test_graph/test_graph_cbd.py:10: note:     def join(__a, Union[bytes, PathLike[bytes]], *paths: Union[bytes, PathLike[bytes]]) -> bytes
  Found 2 errors in 1 file (checked 299 source files)
```

The following should work:

```suggestion
EXAMPLE_GRAPH_FILE_PATH = TEST_DATA_DIR / "spec" / "cbd" / "example_graph.rdf"
EXAMPLE_GRAPH_CBD_FILE_PATH = TEST_DATA_DIR / "spec" / "cbd" / "example_graph_cbd.rdf"
```

**Review by aucampia — COMMENTED** [review]
Looks very good now, just some issue with the Paths.

File: test/test_graph/test_graph_cbd.py [hunk 5]
```diff
@@ -109,3 +122,21 @@ def testCbdReified(get_graph):
     )
 
     assert len(g.cbd(EX.R6)) == (3 + 5 + 5), "cbd() for R6 should return 12 triples"
+
+
+def testCbdExample():
```
Comment (aucampia) [inline test/test_graph/test_graph_cbd.py:127]:
```suggestion
def test_cbd_example():
```

Function names should conform to PEP8[[ref](https://github.com/RDFLib/rdflib/blob/master/docs/developers.rst#introduction)]:

https://peps.python.org/pep-0008/#function-and-variable-names

> Function names should be lowercase, with words separated by underscores as necessary to improve readability.

**Comment by asejal** [comment]
pre-commit.ci autofix

**Review by aucampia — APPROVED** [review]
Thanks @asejal - very happy to have this in place.

**Comment by aucampia** [comment]
Will wait for some more feedback but as this is just adding a test I think it is fine with one review, so I will merge it tomorrow unless there is more feedback.

**Comment by asejal** [comment]
Hi @aucampia, since the pull request has been approved will it be possible for you to merge it? Thanks.

**Review by aucampia — APPROVED** [review]

**Pull request merged**